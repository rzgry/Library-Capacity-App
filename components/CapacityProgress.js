import React from 'react';
import { Bar, Circle } from 'react-native-progress';
import CapacityLimits from '../constants/CapacityLimits';
import Colors from '../constants/Colors';

function progressColour(progress) {
  if (progress <= CapacityLimits.low) {
    return Colors.success;
  }

  if (progress <= CapacityLimits.medium) {
    return Colors.warning;
  }

  return Colors.danger;
}

export const ProgressBar = ({ progress, ...props }) => (
  <Bar color={progressColour(progress)} progress={progress} borderColor="grey" {...props} />
);

ProgressBar.defaultProps = {
  progress: 0,
};

export const ProgressCircle = ({ progress, ...props }) => (
  <Circle color={progressColour(progress)} progress={progress} {...props} />
);

ProgressCircle.defaultProps = {
  progress: 0,
};
