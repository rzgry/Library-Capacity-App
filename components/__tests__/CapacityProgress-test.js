import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { ProgressBar, ProgressCircle } from '../CapacityProgress';

describe('<ProgressBar />', () => {
  it('renders correctly with 0 progress', () => {
    const tree = renderer.create(<ProgressBar progress={0} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with 0 progress', () => {
    const tree = renderer.create(<ProgressBar progress={50} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with 100 progress', () => {
    const tree = renderer.create(<ProgressBar progress={100} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<ProgressCircle />', () => {
  it('renders correctly with 0 progress', () => {
    const tree = renderer.create(<ProgressCircle progress={0} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with 0 progress', () => {
    const tree = renderer.create(<ProgressCircle progress={50} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with 100 progress', () => {
    const tree = renderer.create(<ProgressCircle progress={100} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
