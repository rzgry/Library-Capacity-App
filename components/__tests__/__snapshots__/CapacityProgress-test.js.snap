// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders correctly with 0 progress 1`] = `
<View
  animated={true}
  animationConfig={
    Object {
      "bounciness": 0,
    }
  }
  animationType="spring"
  indeterminate={false}
  onLayout={[Function]}
  progress={0}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "grey",
        "borderRadius": 4,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": 150,
      },
      undefined,
    ]
  }
  useNativeDriver={false}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#5cb85c",
        "height": 6,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": -74,
          },
          Object {
            "scaleX": 0.0001,
          },
        ],
      }
    }
  />
</View>
`;

exports[`<ProgressBar /> renders correctly with 0 progress 2`] = `
<View
  animated={true}
  animationConfig={
    Object {
      "bounciness": 0,
    }
  }
  animationType="spring"
  indeterminate={false}
  onLayout={[Function]}
  progress={50}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "grey",
        "borderRadius": 4,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": 150,
      },
      undefined,
    ]
  }
  useNativeDriver={false}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#d9534f",
        "height": 6,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": 0,
          },
          Object {
            "scaleX": 1,
          },
        ],
      }
    }
  />
</View>
`;

exports[`<ProgressBar /> renders correctly with 100 progress 1`] = `
<View
  animated={true}
  animationConfig={
    Object {
      "bounciness": 0,
    }
  }
  animationType="spring"
  indeterminate={false}
  onLayout={[Function]}
  progress={100}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "grey",
        "borderRadius": 4,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": 150,
      },
      undefined,
    ]
  }
  useNativeDriver={false}
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#d9534f",
        "height": 6,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": 0,
          },
          Object {
            "scaleX": 1,
          },
        ],
      }
    }
  />
</View>
`;

exports[`<ProgressCircle /> renders correctly with 0 progress 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <ARTSurfaceView
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 40,
          "width": 40,
        },
      ]
    }
  >
    <ARTShape
      d={
        Array [
          0,
          20,
          2.5,
          2,
          20,
          2.5,
        ]
      }
      fill={null}
      opacity={1}
      stroke={
        Array [
          0.3607843137254902,
          0.7215686274509804,
          0.3607843137254902,
          1,
        ]
      }
      strokeCap={0}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={3}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    />
    <ARTShape
      d={
        Array [
          0,
          20,
          0.5,
          4,
          20,
          20,
          19.5,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          20,
          20,
          19.5,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={null}
      opacity={1}
      stroke={
        Array [
          0.3607843137254902,
          0.7215686274509804,
          0.3607843137254902,
          1,
        ]
      }
      strokeCap={0}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`<ProgressCircle /> renders correctly with 0 progress 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <ARTSurfaceView
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 40,
          "width": 40,
        },
      ]
    }
  >
    <ARTShape
      d={
        Array [
          0,
          20,
          2.5,
          4,
          20,
          20,
          17.5,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          20,
          20,
          17.5,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={null}
      opacity={1}
      stroke={
        Array [
          0.8509803921568627,
          0.3254901960784314,
          0.30980392156862746,
          1,
        ]
      }
      strokeCap={0}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={3}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    />
    <ARTShape
      d={
        Array [
          0,
          20,
          0.5,
          4,
          20,
          20,
          19.5,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          20,
          20,
          19.5,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={null}
      opacity={1}
      stroke={
        Array [
          0.8509803921568627,
          0.3254901960784314,
          0.30980392156862746,
          1,
        ]
      }
      strokeCap={0}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;

exports[`<ProgressCircle /> renders correctly with 100 progress 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <ARTSurfaceView
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        },
        Object {
          "height": 40,
          "width": 40,
        },
      ]
    }
  >
    <ARTShape
      d={
        Array [
          0,
          20,
          2.5,
          4,
          20,
          20,
          17.5,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          20,
          20,
          17.5,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={null}
      opacity={1}
      stroke={
        Array [
          0.8509803921568627,
          0.3254901960784314,
          0.30980392156862746,
          1,
        ]
      }
      strokeCap={0}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={3}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    />
    <ARTShape
      d={
        Array [
          0,
          20,
          0.5,
          4,
          20,
          20,
          19.5,
          -1.5707963267948966,
          1.5707963267948966,
          1,
          4,
          20,
          20,
          19.5,
          1.5707963267948966,
          -1.5707963267948966,
          1,
          1,
        ]
      }
      fill={null}
      opacity={1}
      stroke={
        Array [
          0.8509803921568627,
          0.3254901960784314,
          0.30980392156862746,
          1,
        ]
      }
      strokeCap={0}
      strokeDash={null}
      strokeJoin={1}
      strokeWidth={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    />
  </ARTSurfaceView>
</View>
`;
